buildall

```diff
@@ -12,6 +12,49 @@ buster
 unstable
 "
 
+DISTS_WITH_SNAPSHOT="buster"
+
+get_latest_month_query(){
+    local -r snapshot_tmp_dir=$(mktemp -d)
+    local -r snapshot_archive_tmp_file="${snapshot_tmp_dir}/archive.html"
+    mkdir -p "${snapshot_tmp_dir}"
+
+    ! curl -sSfL "https://snapshot.debian.org/archive/debian/" > "$snapshot_archive_tmp_file" && echo "Error when accessing https://snapshot.debian.org/archive/debian/" && return 1
+
+    local -r month_query_regex="(\?year=\d\d\d\d&amp;month=\d+)"
+    local -r month_query=$(grep -Po "${month_query_regex}" "${snapshot_archive_tmp_file}" | tail -1)
+
+    [[ -z "$month_query" ]] && echo "Not found snapshots using the following regex: ${month_query_regex}" && return 1
+
+    echo "$month_query"
+}
+
+get_latest_debian_snapshot_id() {
+    local -r snapshot_tmp_dir=$(mktemp -d)
+    local -r snapshot_list_tmp_file="${snapshot_tmp_dir}/month-snapshots.html"
+    mkdir -p "${snapshot_tmp_dir}"
+
+    ! month_query=$(get_latest_month_query) && return 1
+
+    ! curl -sSfL "https://snapshot.debian.org/archive/debian/$month_query" > "$snapshot_list_tmp_file" && echo "Not found snapshots for these parameters: query=${month_query}" && return 1
```

Comment:
why do we check for errors here? the whole script is using `set -e` so that every error is fatal *unless* explicitly ignored

Reply:
you are right! done!